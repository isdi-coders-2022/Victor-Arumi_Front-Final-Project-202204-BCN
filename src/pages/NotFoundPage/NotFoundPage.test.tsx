import { BrowserRouter } from "react-router-dom";
import NotFoundPage from "./NotFoundPage";
import TestRenderer from "react-test-renderer";
const { render, screen } = require("@testing-library/react");

describe("Given a HomePage component", () => {
  describe("When it's invoked)", () => {
    test("Then it should render a the HomePage component with two buttons elements", () => {
      const expectedText = "Ups... p√°gina no encontrada";

      render(
        <BrowserRouter>
          <NotFoundPage />
        </BrowserRouter>
      );

      const selectedText = screen.getByText(expectedText);

      expect(selectedText).toBeInTheDocument();
    });

    test("Then it should always match with this snapshot", () => {
      const NotFoundPageRender = TestRenderer.create(
        <BrowserRouter>
          <NotFoundPage />
        </BrowserRouter>
      );

      expect(NotFoundPageRender).toMatchSnapshot();
    });
  });
});
