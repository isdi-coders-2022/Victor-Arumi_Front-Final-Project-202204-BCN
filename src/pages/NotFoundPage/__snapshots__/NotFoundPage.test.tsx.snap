// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a HomePage Component When it's invoked) Then it should always match with this snapshot 1`] = `
<div
  className="sc-bczRLJ cyiygk"
>
  <h1
    className="text-center"
  >
    Ups... p치gina no encontrada
  </h1>
  <img
    alt="Imagen de p치gina de error"
    src="../../images/404.webp"
  />
</div>
`;

exports[`Given a HomePage component When it's invoked) Then it should always match with this snapshot 1`] = `
<div
  className="sc-bczRLJ cyiygk"
>
  <h1
    className="text-center"
  >
    Ups... p치gina no encontrada
  </h1>
  <img
    alt="Imagen de p치gina de error"
    src="../../images/404.webp"
  />
</div>
`;
